app/scripts/controllers/transactions/tx-state-manager.test.js

```diff
@@ -200,48 +209,49 @@ describe('TransactionStateManager', function () {
 
       const txm = new TxStateManager({
         initState: {
-          transactions: [
-            submittedTx0,
-            unapprovedTx1,
-            approvedTx2,
-            confirmedTx3,
-          ],
+          transactions: {
+            [submittedTx0.id]: submittedTx0,
+            [unapprovedTx1.id]: unapprovedTx1,
+            [approvedTx2.id]: approvedTx2,
+            [confirmedTx3.id]: confirmedTx3,
+          },
         },
         getNetwork: () => currentNetworkId,
         getCurrentChainId: () => currentChainId,
       });
 
-      assert.deepEqual(txm.getTxList(2), [approvedTx2, confirmedTx3]);
+      assert.deepEqual(txm.getTransactions({ limit: 2 }), [
+        approvedTx2,
+        confirmedTx3,
+      ]);
     });
 
     it('should return a list of transactions, limited by N unique nonces when there ARE duplicates', function () {
-      const submittedTx0s = [
-        {
-          id: 0,
-          metamaskNetworkId: currentNetworkId,
-          time: 0,
-          txParams: {
-            from: '0xAddress',
-            to: '0xRecipient',
-            nonce: '0x0',
-          },
-          status: TRANSACTION_STATUSES.SUBMITTED,
+      const submittedTx0 = {
+        id: 0,
+        metamaskNetworkId: currentNetworkId,
+        time: 0,
+        txParams: {
+          from: '0xAddress',
+          to: '0xRecipient',
+          nonce: '0x0',
         },
-        {
-          id: 0,
-          metamaskNetworkId: currentNetworkId,
-          time: 0,
-          txParams: {
-            from: '0xAddress',
-            to: '0xRecipient',
-            nonce: '0x0',
-          },
-          status: TRANSACTION_STATUSES.SUBMITTED,
+        status: TRANSACTION_STATUSES.SUBMITTED,
+      };
+      const submittedTx0Dupe = {
+        id: 1,
```

Comment:
Hmm. I see that this `id` has changed, so this is no longer an exact duplicate transaction. But I can't think of any benefit to us testing two transactions with the same ID - lots of things would already not be working correctly if someone ended up in that state.